extension/js/background.js

```diff
@@ -35,18 +35,33 @@ var salr_client = false;
  * within the extension.
  *
  */
-var port = chrome.runtime.connect();
-
-port.onMessage.addListener(function(data) {
-    if (!salr_client) {
-        salr_client = new SALR(data, chrome.extension.getURL("images/"));
-    }
-});
+var port = createPort();
 
 // Request the settings from the extension
 port.postMessage({'message': 'GetPageSettings'});
 
+// Create a port to communicate through.
+function createPort() {
+    let p = chrome.runtime.connect();
+    p.onMessage.addListener(function(data) {
+        if (!salr_client) {
+            salr_client = new SALR(data, chrome.extension.getURL("images/"));
+        }
+    });
+    p.onDisconnect.addListener(() => {
+        port = null;
+        p = null;
+        salr_client = false;
```

Comment:
Can remove this I think. No matter how many times we re setup the port, the instantiated `SALR` class should be essentially the same as neither `data` nor `chrome.extension.getURL("images/")` should change. Additionally, it leverages the `postMessage` function so it doesn't have an explicit binding to the port we make, so even if the definition for `port` changes, it doesn't really matter.

If we did want to reset `SALR` each time to a fresh instance, would have to do `port.postMessage({'message': 'GetPageSettings'})` as we rely on its response to set `data` for `SALR`, and again, that just seems like wasted overhead.